plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/TenantDataProvider.java

```diff
@@ -179,6 +118,72 @@ private String getResponseBody(final String endpoint, final String keycloakToken
     return response.asString();
   }
 
+  private class TenantDataLoader extends CacheLoader<CacheKey, UserCheTenantData> {
```

Comment:
if we started the refactoring maybe it makes sense to create a separate non-private class ?

Reply:
I only moved it to a private class because the previous format (`CacheLoader#from(...)`) doesn't allow exceptions to be thrown. I still think that these classes (`TenantDataLoader` and `CacheKey`) only many sense within `TenantDataProvider` for now.